examples/v2/htcondor/condor.py

```diff
@@ -0,0 +1,251 @@
+# Copyright 2015 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Generates config for htcondor cluster with 1 master, 1 job submitter
+and n compute nodes.
+"""
+
+
+def GenerateConfig(evaluation_context):
+    return """
+resources:
+- name: condor-network
+  type: compute.v1.network
+  properties:
+    IPv4Range: 10.240.0.0/16
+- name: ssh-firewall-rule
+  type: compute.v1.firewall
+  properties:
+    network: $(ref.condor-network.selfLink)
+    sourceRanges: ["0.0.0.0/0"]
+    allowed:
+    - IPProtocol: TCP
+      ports: ["22"]
+- name: all-internal-firewall-rule
+  type: compute.v1.firewall
+  properties:
+    network: $(ref.condor-network.selfLink)
+    sourceRanges: ["10.240.0.0/16"]
+    allowed:
+    - IPProtocol: TCP
+      ports: ["0-65535"]
+    - IPProtocol: UDP
+      ports: ["0-65535"]
+    - IPProtocol: ICMP
+- name: condor-master
+  type: compute.v1.instance
+  properties:
+    zone: %(zone)s
+    machineType: https://www.googleapis.com/compute/v1/projects/%(project)s/zones/%(zone)s/machineTypes/%(instancetype)s
+    disks:
+    - deviceName: boot
+      type: PERSISTENT
+      boot: true
+      autoDelete: true
+      initializeParams:
+        sourceImage: https://www.googleapis.com/compute/v1/projects/debian-cloud/global/images/debian-8-jessie-v20160606
+    networkInterfaces:
+    - network: $(ref.condor-network.selfLink)
+      accessConfigs:
+      - name: External NAT
+        type: ONE_TO_ONE_NAT
+    serviceAccounts:
+      - email: "default"
+        scopes:
+        - "https://www.googleapis.com/auth/logging.write"
+    tags:
+      items:
+        - condor-master
+    metadata:
+      items:
+        - key: startup-script
+          value: |
```

Comment:
This file is going to look better if you have it in a separate file, and you include in your imports

imports:
- path: startup-script.sh

And here in the python file you can refer as

context.imports['startup-script.sh']

You can even make it a variable if you dont want to hardcode the name as well.